### PR TITLE
Cypress with autogenerated client

### PR DESCRIPTION
**Pull request by nl-kaidzen**
- Switching to an automatically generated client;
- Support for sending custom headers;
- Updated delay timing

### CODE REVIEW

File: qase-cypress/src/index.ts [hunk 1]
```diff
@@ -1,131 +1,230 @@
...
+            /['"\n]+/g,
+            ''
         );
+        const qaseapiVersion = CypressQaseReporter.getPackageVersion('qaseio');
+        const cypressVersion = CypressQaseReporter.getPackageVersion('cypress');
+        const cypressCaseReporterVersion = this.getPackageVersion(
+            'cypress-qase-reporter'
+        );
+        const xPlatformHeader = `node=${nodeVersion}; npm=${npmVersion}; os=${os}; arch=${arch}`;
+        // eslint-disable-next-line max-len
```
Comment (kmalinin) [inline qase-cypress/src/index.ts:155]: no need to disable eslint here

File: qase-cypress/src/index.ts [hunk 2]
```diff
@@ -1,131 +1,230 @@
...
-                while ((this.shouldPublish !== 0) && (Date.now() < endTime)) {
+                while (!this.publishedResults && Date.now() < endTime) {
                     // sleep 500 ms
                     const sharedArrayBuffer = new SharedArrayBuffer(8);
                     const int32array = new Int32Array(sharedArrayBuffer);
                     Atomics.wait(int32array, 0, 0, 500);
                 }
-                if (this.runId && this.shouldPublish !== 0) {
+                if (!this.publishedResults) {
                     this.log(
```
Comment (kmalinin) [inline qase-cypress/src/index.ts:228]: 30 or 60 seconds?

File: qase-cypress/src/index.ts [hunk 3]
```diff
@@ -134,47 +233,76 @@ class CypressQaseReporter extends reporters.Base {
...
+    private async createRun(
+        name: string | undefined,
+        description: string | undefined,
+        cb: (created: IdResponse | undefined) => void
+    ): Promise<void> {
+        try {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const environmentId =
+        Number.parseInt(CypressQaseReporter.getEnv(Envs.environmentId)!, 10) ||
+        this.options.environmentId;
```
Comment (kamilsk) [inline qase-cypress/src/index.ts:260]: please take a look: Forbidden non-null assertion

**Review by kamilsk — APPROVED** [review]

File: qase-cypress/src/index.ts (hunk 1 above)
Comment (nl-kaidzen) [reply qase-cypress/src/index.ts]: Fixed

File: qase-cypress/src/index.ts (hunk 2 above)
Comment (nl-kaidzen) [reply qase-cypress/src/index.ts:227]: Fixed 30 sec

File: qase-cypress/src/index.ts [hunk 4]
```diff
@@ -214,10 +213,10 @@ class CypressQaseReporter extends reporters.Base {
                 this.log('Nothing to send.');
             } else if (this.runId) {
                 this.log(
-                    chalk`{blue Waiting for 60 seconds to publish pending results}`
+                    chalk`{blue Waiting for 30 seconds to publish pending results}`
                 );
 
-                const endTime = Date.now() + 60e3;
+                const endTime = Date.now() + 30e3;
```
Comment (kamilsk) [inline qase-cypress/src/index.ts:219]: Please make it configurable

**Review by kamilsk — CHANGES_REQUESTED** [review]

File: qase-cypress/src/index.ts [hunk 5]
```diff
@@ -225,7 +230,7 @@ class CypressQaseReporter extends reporters.Base {
                 }
                 if (!this.publishedResults) {
                     this.log(
-                        chalk`{red Could not send all results for 30 seconds after run. Please contact Qase Team.}`
+                        chalk`{red Could not send all results for ${delay} seconds after run. Please contact Qase Team.}`
```
Comment (kamilsk) [inline qase-cypress/src/index.ts:234]: did you see image

**Review by kamilsk — APPROVED** [review]

**Pull request merged**